ntbs-service/Pages/Notifications/LinkedNotifications.cshtml.cs

```diff
@@ -20,18 +20,20 @@ public async Task<IActionResult> OnGetAsync(int id)
             {
                 return NotFound();
             }
-            NotificationBannerModel = new NotificationBannerModel(Notification);
 
-            await GetLinkedNotifications();
+            await AuthorizeAndSetBannerAsync();
+
             NotificationId = Notification.NotificationId;
+            await GetLinkedNotifications();
 
             if (Group == null)
             {
                 return NotFound();
             }
 
             LinkedNotifications = Group.Notifications.Where(n => n.NotificationId != NotificationId)
-                .Select(n => NotificationBannerModel.WithLink(n)).ToList();
+                .Select(async n => NotificationBannerModel.WithLink(n, await authorizationService.CanEdit(User, n)))

```

Comment:
the async-await stuff got pretty unwieldy here.
I think the `WithLink` method was quite nice when it was simple, but perhaps it's outgrowing its usecase here.
Not sure what the right format would be, but I reckon this piece of code could use a bit of massaging  to get clearer :)

Reply:
I'll have a think about how this could be refactored!